src/module/hooks/combat.mjs

```diff
@@ -270,3 +283,30 @@ async function renderOngoingEffectsCard(title, combatant, effectData) {
     };
     ChatMessage.create(chatData, {});
 }
+
+async function executeLifecycleMacro(combatant, hookName) {
+    // If this isn't the actor's player, emit a socket request for that player to execute the hook
+    if (game.user?.character?.id !== combatant.actor.id) {
```

Comment:
My GM client received an error on this, though I would have expected it work due to the optional chaining.

```
Uncaught (in promise) TypeError: game.user.character is null
    _handleActorLifecycleHook archmage.js:1694
```

One question I have is what happens if multiple users are logged in and have different character IDs? Does the socket fire several times for the client of the user that does have a matching character?

Reply:
I think this only fires on the client where the "next turn" button is clicked, but I'd need to confirm that. The exception is weird, the `?.`'s should prevent that.

Reply:
Yup, looks like the hook only fires on the client where "next turn" is clicked, so the player's client will only receive this once.


https://github.com/user-attachments/assets/e76d72cd-2880-4a5b-81c1-f3889f8f23ab